domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt

```diff
@@ -201,6 +201,7 @@ class QuestionAssessmentProgressController @Inject constructor(
           // in an 'always showing hint' situation. This can specifically happen if hint throws an
           // exception.
           progress.advancePlayStageTo(TrainStage.VIEWING_STATE)
+          progress.hintViewed()
```

Comment:
I am wondering if we should put this in the try block rather than the finally block. Reason being that we should penalize the user only if they actually were able to view the hint. If the hint fails to be revealed for whatever reason, we probably shouldn't penalize them. What do you think?

Reply:
Yeah that makes total sense, thanks for pointing that out! I just double checked this, I assumed (incorrectly) that the finally block only executes if you don't get an exception.

Reply:
Fixed, thanks!